hr_holidays_public/models/resource_calendar.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2017-2018 Tecnativa - Pedro M. Baeza
+# Copyright 2018 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.addons.resource.models.resource import Intervals
+
+from pytz import timezone
+from datetime import datetime, time
+from dateutil import rrule
+
+
+class ResourceCalendar(models.Model):
+    _inherit = 'resource.calendar'
+
+    def _public_holidays_leave_intervals(
+        self,
```

Comment:
I have to say I don't like too much this style for declaring methods, as it makes very long the signature without clear distinction over the rest of the code. I prefer this style:
```python
def _public_holidays_leave_intervals(self, start_dt, end_dt, employe_id, tz,
                                     extra_argument_1, extra_argument_2):
```
I even prefer it for the diff, because although it requires 2 diff lines, you see clearly that is a change on the arguments instead other kind of change.